parquet-hadoop/src/main/java/parquet/hadoop/ParquetRecordReader.java

```diff
@@ -135,7 +135,18 @@ public void initialize(InputSplit inputSplit, TaskAttemptContext context)
               + context.getClass().getCanonicalName());
     }
 
-    initializeInternalReader((ParquetInputSplit)inputSplit, ContextUtil.getConfiguration(context));
+    // ensure all splits are ParquetInputSplits
+    ParquetInputSplit parquetSplit;
+    if (inputSplit instanceof ParquetInputSplit) {
+      parquetSplit = (ParquetInputSplit) inputSplit;
+    } else if (inputSplit instanceof FileSplit) {
+      parquetSplit = ParquetInputSplit.from((FileSplit) inputSplit);
+    } else {
+      throw new IllegalArgumentException(
+          "Invalid split (not a FileSplit or ParquetInputSplit): " + inputSplit);
+    }
+
+    initializeInternalReader(parquetSplit, ContextUtil.getConfiguration(context));
```

Comment:
ParquetInputSplit extends FileSplit so we probably can just depend on FileSplit.
On the other end we should really receive only what was returned by getSplits which always returns ParquetInputSplits in your implementation.
In both cases the logic above can be simplified.